### PR TITLE
Bump kubernetes version to `1.27.2` and remove fork of apiserver

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
Bump the apiserver to k8s 1.27.

### Motivation
We currently use a fork on the Datadog Agent for the k8s.io/apiserver 

This limits our ability to easily bump the Kubernetes dependency and potentially introduces differences when our Go modules are used externally, where replace statements are ignored.

### Additional Notes
https://github.com/DataDog/datadog-agent/pull/16652 needs to be merged first

### Possible Drawbacks / Trade-offs
When we bump Kubernetes version, it often has side effects. Some metrics can disappear on some specific versions of kubernetes. It is unclear how to evaluate that.

### Describe how to test/QA your changes
All tests should pass.
Verify that all basic functionalities using Kubernetes work.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/use-leases-leader-election-c278c71fb39c2c84.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+enhancements:
+  - |
+    Use ``leases.coordination/v1`` for cluster-agent leader election when they are available.
```
Comment (rtrieu) [inline releasenotes/notes/use-leases-leader-election-c278c71fb39c2c84.yaml:4]:
```suggestion
    Use ``leases.coordination/v1`` for cluster-agent leader election when it becomes available.
```

**Review by rtrieu — APPROVED** [review]
Minor feedback, but approved.

File: releasenotes/notes/use-leases-leader-election-c278c71fb39c2c84.yaml (hunk 1 above)
Comment (rtrieu) [reply releasenotes/notes/use-leases-leader-election-c278c71fb39c2c84.yaml:4]: @AliDatadog Updated my suggestion!

File: pkg/collector/corechecks/cluster/helm/helm.go
Comment (AliDatadog) [inline pkg/collector/corechecks/cluster/helm/helm.go:1]: Not sure if an error should be returned here

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file (the copyright yml file)

**Review by clamoriniere — APPROVED** [review]
please create a card for implementing the follow up 🙇 

Could you also review which Resouces haven been deprecated/removed since kube client 1.25. 
thanks

**Review by fisherevans — APPROVED** [review]

**Pull request merged**